sqlgen/db.go

```diff
@@ -361,6 +362,68 @@ func (db *DB) InsertRow(ctx context.Context, row interface{}) (sql.Result, error
 	return db.execWithTrace(ctx, query, "InsertRow")
 }
 
+// InsertRows inserts multiple rows into the database, chunksize rows at a time.
+// Most SQL db enforce a limit on max size of packet, which is why we need to break
+// the rows into chunks.
+//
+// rows should be an array of pointers to a struct, for example:
+//
+//   user1 := &User{Name: "foo"}
+//   user2 := &User{Name: "fan"}
+//   if err := db.InsertRows(ctx, [](*User){user1, user2}, 100); err != nil {
+//
+func (db *DB) InsertRows(ctx context.Context, rows interface{}, chunkSize int) error {
+	val := reflect.ValueOf(rows)
+	kind := val.Kind()
+	if kind != reflect.Slice && kind != reflect.Array {
+		return fmt.Errorf("expect array/slice got %s", val.Kind().String())
+	}
+
+	rowsData := make([]interface{}, val.Len())
+	for i := 0; i < val.Len(); i++ {
+		rowsData[i] = val.Index(i).Interface()
+	}
+
+	var tx *sql.Tx
+	if !db.HasTx(ctx) {
+		var err error
+		ctx, tx, err = db.WithTx(ctx)
+		if err != nil {
+			return err
+		}
+		defer tx.Rollback()
+	}
+
+	for j := 0; j < len(rowsData); j += chunkSize {
+		sliceLength := chunkSize
+		if len(rowsData) < j+chunkSize {
+			sliceLength = len(rowsData) - j
+		}
+		slice := rowsData[j:sliceLength]
```

Comment:
should this be `rowsData[j:(j+sliceLength)]`? if so, will you add a unit test for this?

nit: maybe this is my personal style, i find the inverted logic here more readable, like this
```
func (db *DB) InsertRows(ctx context.Context, rows interface{}, maxChunkSize int) error {
  ...
  chunkSize := len(rowsData) - j
  if chunkSize > maxChunkSize {
    chunkSize = maxChunkSize
  }
  chunk := rowsData[j:j+chunkSize]
  ...
}
```

Reply:
good catch. The existing unit test already tests this path. I am surprised to see unit tests pass. Let me investigate

Reply:
@changpingc fixed. Also, added a unit test with small chunk size

Reply:
where is this unit test?